### PR TITLE
fix(metrics): separate high watermark and consumer lag

### PR DESCRIPTION
**Pull request by yuhao-su**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

The original one in misleading. Make people think high watermark reflects consumer lag

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: grafana/risingwave-dev-dashboard.dashboard.py [hunk 1]
```diff
@@ -743,16 +743,26 @@ def section_streaming(outer_panels):
...
                     [
                         panels.target(
                             f"{metric('source_kafka_high_watermark')}",
-                            "source={{source_id}} partition={{partition}}",
+                            "high watermark: source={{source_id}} partition={{partition}}",
                         ),
                         panels.target(
                             f"{metric('source_latest_message_id')}",
-                            "source={{source_id}} partition={{partition}} actor_id={{actor_id}}",
+                            "latest msg: source={{source_id}} partition={{partition}} actor_id={{actor_id}}",
```
Comment (hzxa21) [inline grafana/risingwave-dev-dashboard.dashboard.py:755]: I think it is better to move these two metrics to other section and only keep the Consumer Lag metric under streaming avoid unnecessarily loading detailed info

File: grafana/risingwave-dev-dashboard.dashboard.py [hunk 2]
```diff
@@ -743,16 +743,26 @@ def section_streaming(outer_panels):
...
+                        ),
+                    ],
+                ),
+                panels.timeseries_count(
+                    "Kafka Consumer Lag Size",
+                    "Kafka Consumer Lag Size by source_id, partition and actor_id",
+                    [
+                        panels.target(
+                            f"{metric('source_kafka_high_watermark')} - on(source_id, partition) group_right() {metric('source_latest_message_id')}",
+                            "source={{source_id}} partition={{partition}}",
```
Comment (hzxa21) [inline grafana/risingwave-dev-dashboard.dashboard.py:765]: There can be multiple actors in different streaming job consuming the same source in different speed so the legend should be `source={{source_id}} partition={{partition}} actor = {{actor_id}}`

File: grafana/risingwave-dev-dashboard.dashboard.py [hunk 3]
```diff
@@ -743,16 +743,26 @@ def section_streaming(outer_panels):
...
+                            "latest msg: source={{source_id}} partition={{partition}} actor_id={{actor_id}}",
+                        ),
+                    ],
+                ),
+                panels.timeseries_count(
+                    "Kafka Consumer Lag Size",
+                    "Kafka Consumer Lag Size by source_id, partition and actor_id",
+                    [
+                        panels.target(
+                            f"{metric('source_kafka_high_watermark')} - on(source_id, partition) group_right() {metric('source_latest_message_id')}",
```
Comment (hzxa21) [inline grafana/risingwave-dev-dashboard.dashboard.py:764]: Due to misalignment and non-atomicity of metrics reporting, it is possible to have source_latest_message_id > source_kafka_high_watermark. Therefore, I recommend cap the result to 0:
```
f"{clamp_min(metric('source_kafka_high_watermark')} - on(source_id, partition) group_right() {metric('source_latest_message_id')}, 0)"
```

File: grafana/risingwave-dev-dashboard.dashboard.py (hunk 1 above)
Comment (yuhao-su) [reply grafana/risingwave-dev-dashboard.dashboard.py:755]: Any suggestion to which section?

**Comment by yuhao-su** [comment]
> I hoped to do some joins on the grafana on the same source_id when inpl the metric. But grafana cannot do this kind of join, so I put the two metrics together to express the lag 😞

The "join" in PromSql is called `group_left/group_right`

**Review by hzxa21 — APPROVED** [review]

**Pull request merged**